src/System.Private.ServiceModel/src/System/ServiceModel/Security/SecurityUtils.cs

```diff
@@ -793,16 +794,33 @@ static internal bool IsDefault(NetworkCredential credential)
 
             internal static string GetCurrentUserIdAsString(NetworkCredential credential)
             {
-#if SUPPORTS_WINDOWSIDENTITY
-                using (WindowsIdentity self = WindowsIdentity.GetCurrent())
+                if (!string.IsNullOrEmpty(currentUser))
                 {
-                    return self.User.Value;
+                    return currentUser;
+                }
+
+                // CurrentUser could be set muliple times
+                // This is fine because it does not affect the value returned.
+#if SUPPORTS_WINDOWSIDENTITY // NegotiateStream
+                try
+                {
+                    using (WindowsIdentity self = WindowsIdentity.GetCurrent())
+                    {
+                        currentUser = self.User.Value;
+                    }
+                }
+                catch (PlatformNotSupportedException)
+                {
+                    //WindowsIdentity is not supported on *NIX
+                    //so returning a username which is very unlikely to be a real username;
+                    currentUser = "____CURRENTUSER_NOT_AVAILABLE____";
                 }
 #else
-                // There's no way to retrieve the current logged in user Id in UWP apps or in
-                // *NIX so returning a username which is very unlikely to be a real username;
-                return "____CURRENTUSER_NOT_AVAILABLE____";
-#endif
+                // There's no way to retrieve the current logged in user Id in UWP apps
+                currentUser = "____CURRENTUSER_NOT_AVAILABLE____";
```

Comment:
>"____CURRENTUSER_NOT_AVAILABLE____" [](start = 30, length = 35)

Nit: this could be a private const in this class to avoid duplication.   Not blocking.